src/openrct2/actions/ParkSetParameterAction.hpp

```diff
@@ -0,0 +1,97 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../interface/Window.h"
+#include "../ride/ShopItem.h"
+#include "../world/Park.h"
+#include "GameAction.h"
+
+enum class ParkParameter : uint8_t
+{
+    Close,
+    Open,
+    SamePriceInPark,
+    Count
+};
+
+DEFINE_GAME_ACTION(ParkSetParameterAction, GAME_COMMAND_SET_PARK_OPEN, GameActionResult)
+{
+private:
+    uint8_t _parameter;
```

Comment:
Can you assign ParkParameter::Count as the default value, if serialisation fails it will be at least not uninitialized and Query can can fail happily.